### PR TITLE
[wip] Leverage clients provided by go-mod-core-contracts

### PR DESCRIPTION
**Pull request by couralex6**
Started using the go-mod-core-contract clients instead of implementing the rest calls from scratch.

Note:
* Some of the clients are not implemented in go-mod-core-contract. (Addressable list, ds list/rm, ...)
* Not using the registry at this point (future work)

Signed-off-by: Alex Courouble <acourouble@vmware.com>

### CODE REVIEW

File: cmd/device/rm/rm.go [hunk 1]
```diff
@@ -38,22 +41,40 @@ You can use: '$ edgex device list' to find a device's name and ID.`,
...
+				),
+			)
+
+			//devices, err := mdc.Devices(ctx)
+
+			err := mdc.DeleteByName(ctx, deviceID)
 
 			if err != nil {
-				fmt.Println(err)
+				fmt.Printf("Removed: %s\n", deviceID)
```
Comment (mkbhanda) [inline cmd/device/rm/rm.go]: Is this not a delete failure case? Believe you wanted to test for == nil

Comment (couralex6) [reply cmd/device/rm/rm.go]: You're right. it should be `err == nil`. We're checking if `DeleteByName` was successful. If not, we try deleting by ID.

**Comment by couralex6** [comment]
Note: the commands not using the client provided by go-mod-core-contract contain a lot of hard coded value and strings (URLs and port numbers).

File: cmd/deviceservice/rm/rm.go [hunk 2]
```diff
@@ -38,9 +37,9 @@ func NewCommand() *cobra.Command {
 
 			deviceID := args[0]
 			respBody, err := client.DeleteItem(deviceID,
-				config.Conf.MetadataService.DeviceServiceByIDRoute,
-				config.Conf.MetadataService.DeviceServiceBySlugNameRoute,
-				config.Conf.MetadataService.Port)
+				"deviceservice/id/",
+				"deviceservice/name/",
+				"48081")
```
Comment (mkbhanda) [inline cmd/deviceservice/rm/rm.go]: odd that this call needed a port number

File: cmd/subscription/rm/rm.go [hunk 3]
```diff
@@ -36,9 +35,9 @@ func removeSubscriptionHandler(cmd *cobra.Command, args []string) {
 	subscriptionlID := args[0]
 
 	respBody, err := client.DeleteItem(subscriptionlID,
-		config.Conf.NotificationService.SubscriptionByIDRoute,
-		config.Conf.NotificationService.SubscriptionByNameSlugRoute,
-		config.Conf.NotificationService.Port)
+		"subscription",
+		"subscription/name/",
+		"48060")
```
Comment (mkbhanda) [inline cmd/subscription/rm/rm.go]: Saw 48060 in the context of meta service

File: cmd/deviceservice/rm/rm.go (hunk 2 above)
Comment (couralex6) [reply cmd/deviceservice/rm/rm.go]: Why so? client.DeleteItem requires the port to build the URL

File: cmd/subscription/rm/rm.go (hunk 3 above)
Comment (mkbhanda) [reply cmd/subscription/rm/rm.go]: I missed that! Thanks for clarifying

**Pull request merged**